wp-content/themes/hurumap/functions.php

```diff
@@ -218,23 +218,33 @@ function get_base_url() {
     }
 }
 
+
+function get_token(){
+    $request = new WP_REST_Request( 'POST', '/jwt-auth/v1/token' );
+    $request->set_body_params( [ 'username' => constant("WP_DEFAULT_USERNAME"),"password"=>constant("WP_DEFAULT_PASSWORD") ] );
+    $response = rest_do_request( $request );
+    $server = rest_get_server();
+    $data = $server->response_to_data( $response, false );
```

Comment:
1. Anything we can do to gracefully handle any errors?
2. How long is the token return here active for? We should set it to < 30 seconds and none renewable if possible i.e. generate new token every time so that this doesn't become a security risk since we're passing it to FE as a query param.

Reply:
@kilemensi  erm, now, errors are handled gracefully on the frontend. if it doesn't get the token, it redirects to 404

Reply:
Sure @KobbyMmo ... I meant on WP side. By the time we show 404 on FE side, the user might not know what the issue was and may think their draft page/post is lost or something.

Reply:
From our discussions :

  1. We've added custom error page on the FE in case a token is missing, and
  2. `$data = $server->response_to_data( $response, false );` and `$data["token"];` will not throw any errors and hence there is no need to add error handling here.
  
  If the above is correct @KobbyMmo , resolve this conversation.